### ISSUE TITLE
Error when import github_repository resource in child module

### ISSUE THREAD

**Issue by Ameausoone** [terraform-providers/terraform-provider-github#283]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Terraform v0.12.9
+ provider.github v2.2.1

### Affected Resource(s)
Please list the resources as a list, for example:
- github_repository

### Terraform Configuration Files

github-terraform-module.tf 
```hcl
variable "github_token" {
  type = "string"
}
variable "github_organization" {
  default = "myorganization"
}
provider "github" {
  token = "${var.github_token}"
  organization = "${var.github_organization}"
}
variable "name" {
  default = "terraform-test-repo"
}
resource "github_repository" "projects" {
  name = var.name
  private = true
}
```

main project => main-terraform-files.tf
```hcl
// Set through env var TF_VAR_github_token
variable "github_token" {
  default = ""
}
module "projects" {
  source = "git@github.com:myorganisation/github-terraform-module.git"
  github_token = var.github_token
}
```

### Debug Output

https://gist.github.com/Ameausoone/a140c7b9aacd06575f3843e454fbe6ab

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior

```bash
terraform import 'module.projects.github_repository.projects' terraform-test-repo

github_repository.projects: Importing from ID "terraform-test-repo"...
github_repository.projects: Import prepared!
  Prepared github_repository for import
github_repository.projects: Refreshing state... [id=terraform-test-repo]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

### Actual Behavior

```bash
terraform import 'module.projects.github_repository.projects' terraform-test-repo

Error: If `anonymous` is false, `token` is required.

  on .terraform/modules/projects/github-provider.tf line 9, in provider "github":
   9: provider "github" {

```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform import 'module.projects.github_repository.projects' terraform-test-repo`

### Important Factoids

* If I use directly the module project, and import resource, it works. 
```
terraform import github_repository.projects 'terraform-test-repo'
```

* If I use module, and use "GITHUB_TOKEN" env var to configure github provider, it works.

**Comment by techdragon** [comment]
Still broken in 2.6.1 :-(

It seems as though even if you set the `GITHUB_TOKEN`environment variable, you will need to _remove_ (or comment out) the token attribute in the provider definition, otherwise this doesn't work. 

my import testing results for easier reference: 
token defined in provider = Fail
token defined in provider + `GITHUB_TOKEN` environment variable = Fail
`GITHUB_TOKEN` environment variable = Pass

**Comment by josjaf** [comment]
I'm ahving this issue with 
Terraform v0.13.0
+ provider registry.terraform.io/-/github v2.9.2

**Comment by jcudit** [comment]
This may have been fixed in [`v3.0.0`](https://github.com/terraform-providers/terraform-provider-github/blob/master/CHANGELOG.md#300-september-08-2020) when there were changes to provider configuration.  Pleae re-open if this is still an issue.
